### ISSUE TITLE
Determine how to run openlibrary locally with just python3

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#2293]
### Is your feature request related to a problem? Please describe.
Right now it's difficult to test our python3 support since locally the site can only be run with python2.

### Describe the solution you'd like
Need an easy way to run with python3 (probably 3.7). Ok if a docker rebuild is required.

### Stakeholders
@salman-bhai @cclauss @hornc

**Comment by cclauss** [comment]
We already have automated tests that configure a Python 3 instance whenever you want.  My recommendation would be focus first on removing this line from our codebase:
https://github.com/internetarchive/openlibrary/blob/master/.travis.yml#L17

**Comment by cdrini** [comment]
I think the idea is to avoid having to push to GitHub and wait for Travis every time we want to test with python3? Or is there another way?
What would be a good first step @salman-bhai to contribute to the Python 3 migration?

**Comment by cclauss** [comment]
> focus first on removing this line from our codebase

Writing that led me to #2295 which improves the situation quite a bit.

I do struggle with how you do things _locally_ on this project given that the main repo symlinks (a stale copy) of the separate infogami repo.  It is code in this second repo that is now haltin our Py3 progress.  @hornc

So the trick in 2295 is that you need both Python 3 for real computing and then you also need a Python 2 for npm/node to live it.  [ _Nodejs still requires legacy Python but I have been working with that team to fix that and we are close._ ]  For some reason npm could not find the Python 2.7.15 that is in the bionic image so 2295 does a __pyenv install 2.7.14__ to get around that.

**Comment by cclauss** [comment]
@cdrini Could you please go at this again?  We could really use this now.

**Comment by cdrini** [comment]
@cclauss Could you summarize the next steps on this? (Note our docker is using xenial, not bionic)

**Comment by cclauss** [comment]
I believe that your original problem description at the top of this issue is still the case.  Which distro is not super important for this issue.  What is important is that a contributor can try to run OL on Py3 on their local machine.

**Comment by guyjeangilles** [comment]
From the last community where we talked about this, Infogami was the blocker. Is that still the case @cclauss ?

**Comment by cclauss** [comment]
Yes. Infogami fails on Python 3
https://travis-ci.org/internetarchive/infogami/builds/615095537

**Comment by hornc** [comment]
I have a working test image pushed to the branch https://github.com/internetarchive/openlibrary/tree/python3  

It uses Python 3.5. A proper version should upgrade to Bionic as well.

Still working through the basic code issues though. Having a local image should speed up the process though!

**Comment by cdrini** [comment]
^ here's the diff: https://github.com/internetarchive/openlibrary/compare/master...python3

**Comment by hornc** [comment]
This python3 branch enables devs to run Python3 locally https://github.com/internetarchive/openlibrary/tree/python3

The branch code is still in a state of flux, but @cclauss' PR #2860 demonstrates that the Python can be updated to 3.8

Devs working on Python should be able to use this Docker image to test both Open Library and Infogami interacting in a Python 3 environment.

It is not exactly stable, but the code is changing rapidly anyway. The branch demonstrates the approach and possibility. Anyone wanting to work on Python 3 can use this as a basis and raise issues as they encounter them, ideally attached to existing Python3 issues or PRs.

Closing as completed / unblocked.

**Comment by cclauss** [comment]
Nice progress here!  Thanks much.
